packages/store-sync/src/postgres/tables.ts

```diff
@@ -0,0 +1,44 @@
+import { boolean, index, pgSchema, primaryKey } from "drizzle-orm/pg-core";
+import { transformSchemaName } from "./transformSchemaName";
+import { asAddress, asBigInt, asHex, asNumber } from "./columnTypes";
+
+const schemaName = transformSchemaName("mud");
+
+/**
+ * Singleton table for the state of the chain we're indexing
+ */
+const chainTable = pgSchema(schemaName).table("chain", {
+  chainId: asNumber("chain_id", "bigint").notNull().primaryKey(),
+  lastUpdatedBlockNumber: asBigInt("last_updated_block_number", "numeric"),
+});
+
+const recordsTable = pgSchema(schemaName).table(
+  "records",
+  {
+    address: asAddress("address").notNull(),
+    tableId: asHex("table_id").notNull(),
+    /**
+     * `keyBytes` is equivalent to `abi.encodePacked(bytes32[] keyTuple)`
+     */
+    keyBytes: asHex("key_bytes").notNull(),
+    key0: asHex("key0"),
+    key1: asHex("key1"),
+    staticData: asHex("static_data"),
+    encodedLengths: asHex("encoded_lengths"),
+    dynamicData: asHex("dynamic_data"),
+    isDeleted: boolean("is_deleted"),
+    lastUpdatedBlockNumber: asBigInt("last_updated_block_number", "numeric"),
+  },
+  (table) => ({
+    pk: primaryKey(table.address, table.tableId, table.keyBytes),
```

Comment:
if the primary key is the combination of these three columns, do we need individual indices on `address` and `tableId` for efficient querying on those (where we don't include the `keyBytes` in the query)?

Reply:
the indexes below this handle that and I believe that an index on `[address, tableId, key0]` can still benefit a query using `address=... AND tableId=...`